pkg/gofr/http/responder.go

```diff
@@ -35,9 +36,11 @@ func (r Responder) Respond(data interface{}, err error) {
 
 		return
 	default:
-		resp = response{
-			Data:  v,
-			Error: errorObj,
+		resp = response{Data: data, Error: errorObj}
+
+		// handling where an interface contains a nullable type with a nil value.
+		if isNil(data) {
+			resp = response{Error: errorObj}
 		}
```

Comment:
This code would be simpler I think
	
```suggestion
		// handling where an interface contains a nullable type with a nil value.
		if isNil(data) {
			data = nil
 		}
		resp = response{Data: data, Error: errorObj}
```